-1 
Tested at: 9d12e734b584da98b9ecafced69fb17639a6c73a
 I found an error when building: 
 
 <pre>>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-07-2300/src/FastSimulation/Tracking/plugins/TrackCandidateProducer.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-07-2300/src/FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-07-2300/src/Utilities/StaticAnalyzers/src/EDMPluginDumper.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-07-2300/src/Utilities/StaticAnalyzers/src/FiniteMathChecker.cc 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-07-2300/src/FastSimulation/Tracking/plugins/TrackCandidateProducer.cc: In member function 'virtual void TrackCandidateProducer::produce(edm::Event&, const edm::EventSetup&)':
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-07-2300/src/FastSimulation/Tracking/plugins/TrackCandidateProducer.cc:158:31: error: unused variable 'lastSeedHit' [-Werror=unused-variable]
     const FastTrackerRecHit * lastSeedHit = recHitCandidates.back().hit();
                               ^
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-07-2300/src/FastSimulation/Tracking/plugins/TrackCandidateProducer.cc:165:24: error: 'lastSeedHit' was not declared in this scope
    if(seed.nHits()==0||lastSeedHit->sameId(currentTrackerHit.hit()))     
                        ^
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13214/11070/summary.html

--- a/FastSimulation/Tracking/plugins/TrackCandidateProducer.cc
+++ b/FastSimulation/Tracking/plugins/TrackCandidateProducer.cc
@@ -157,7 +157,7 @@ TrackCandidateProducer::produce(edm::Event& e, const edm::EventSetup& es) {
 
     for (const auto & _hit : recHitCombination) {
       TrajectorySeedHitCandidate currentTrackerHit=TrajectorySeedHitCandidate(_hit.get(),trackerGeometry.product(),trackerTopology.product());
-      if(seed.nHits==0)passedLastSeedHit=true;
+      if(seed.nHits()==0)passedLastSeedHit=true;
 
       if(!passedLastSeedHit)
 	{

--- a/FastSimulation/Tracking/plugins/TrackCandidateProducer.cc
+++ b/FastSimulation/Tracking/plugins/TrackCandidateProducer.cc
@@ -154,14 +154,14 @@ TrackCandidateProducer::produce(edm::Event& e, const edm::EventSetup& es) {
       recHitCandidates.push_back(seedHit);
     }
     bool passedLastSeedHit = false;
-    if(seed.nHits()!=0)
-    const FastTrackerRecHit * lastSeedHit = recHitCandidates.back().hit();
 
     for (const auto & _hit : recHitCombination) {
       TrajectorySeedHitCandidate currentTrackerHit=TrajectorySeedHitCandidate(_hit.get(),trackerGeometry.product(),trackerTopology.product());
+      if(seed.nHits==0)passedLastSeedHit=true;
 
       if(!passedLastSeedHit)
 	{
+	  const FastTrackerRecHit * lastSeedHit = recHitCandidates.back().hit();
 	  if(seed.nHits()==0||lastSeedHit->sameId(currentTrackerHit.hit()))     
 	    {
 	      passedLastSeedHit=true;

